App fails to launch in some devices

Tested app loading in multiple devices, thanks to Browserstack.
In most devices with android version v12.0, it fails to initialize with these error:

```
08-15 07:01:00.437 E/NodeApplicationBootstrapFacade(11656): Bootstrap: Timeout job cancelled for stage: Launching Tor daemon...
08-15 07:01:00.437 E/NodeApplicationBootstrapFacade(11656): kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@3f313c8
08-15 07:01:00.455 E/AndroidApplicationService(11656): networkService.initialize() failed
08-15 07:01:00.455 E/AndroidApplicationService(11656): java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: No virtual method lines()Ljava/util/stream/Stream; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /apex/com.android.art/javalib/core-oj.jar)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:278)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:284)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1629)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.lang.Thread.run(Thread.java:920)
08-15 07:01:00.455 E/AndroidApplicationService(11656): Caused by: java.lang.NoSuchMethodError: No virtual method lines()Ljava/util/stream/Stream; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /apex/com.android.art/javalib/core-oj.jar)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.tor.TorService$$ExternalSyntheticApiModelOutline0.m(R8$$SyntheticClass:0)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.tor.TorService.readExternalTorConfigMap(TorService.java:328)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.tor.TorService.initialize(TorService.java:92)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.node.transport.TorTransportService.initialize(TorTransportService.java:52)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.ServiceNode.getInitializedDefaultNode(ServiceNode.java:270)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.ServiceNodesByTransport.lambda$getInitializedDefaultNodeByTransport$1(ServiceNodesByTransport.java:142)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.ServiceNodesByTransport.$r8$lambda$y4pBO1vy7bTFNomdmryLcUGipuM(ServiceNodesByTransport.java:0)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.ServiceNodesByTransport$$ExternalSyntheticLambda21.get(R8$$SyntheticClass:0)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1627)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	... 3 more
08-15 07:01:00.455 E/AndroidApplicationService(11656): Initializing applicationService failed
08-15 07:01:00.455 E/AndroidApplicationService(11656): java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: No virtual method lines()Ljava/util/stream/Stream; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /apex/com.android.art/javalib/core-oj.jar)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:278)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:284)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1629)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.lang.Thread.run(Thread.java:920)
08-15 07:01:00.455 E/AndroidApplicationService(11656): Caused by: java.lang.NoSuchMethodError: No virtual method lines()Ljava/util/stream/Stream; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /apex/com.android.art/javalib/core-oj.jar)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.tor.TorService$$ExternalSyntheticApiModelOutline0.m(R8$$SyntheticClass:0)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.tor.TorService.readExternalTorConfigMap(TorService.java:328)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.tor.TorService.initialize(TorService.java:92)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.node.transport.TorTransportService.initialize(TorTransportService.java:52)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.ServiceNode.getInitializedDefaultNode(ServiceNode.java:270)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.ServiceNodesByTransport.lambda$getInitializedDefaultNodeByTransport$1(ServiceNodesByTransport.java:142)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.ServiceNodesByTransport.$r8$lambda$y4pBO1vy7bTFNomdmryLcUGipuM(ServiceNodesByTransport.java:0)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at bisq.network.p2p.ServiceNodesByTransport$$ExternalSyntheticLambda21.get(R8$$SyntheticClass:0)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1627)
08-15 07:01:00.455 E/AndroidApplicationService(11656): 	... 3 more
08-15 07:01:00.456 E/NodeApplicationBootstrapFacade(11656): Bootstrap: Timeout job cancelled for stage: Bootstrap to P2P network
```

Devices with this issue: (model / android version)
 - Huawei Nova 11 SE / v12.0
 - Oppo A96 / v12.0
 - Vivo Y22 / v12.0
 - Redmi Note 12 Pro / v12.0
 - Pixel 6 / v12.0
 - Pixel 6 Pro / v12.0
 - Galaxy S22 / v12.0
 
Devices where it runs good: (model / android version)
 - Realme 11X 5G / v13.0
 - Realme 12+ / v14.0
 - Oppo Reno 8T 5G / v14.0
 - Oppo A78 / v13.0
 - Edge 40 Neo / v13.0
 - Edge 50 Fusion / v14.0
 - Edge 60 Fusion / v15.0
 - Vivo Y100 v13.0
 - Vivo V30 / v14.0
 - Vivo T4x 5G / v15.0
 - Redmi Note 12 4G / v13.0
 - Redmi Note 13 Pro 5G / v14.0
 - Redmi Note 14 Pro 5G / v15.0
 - OnePlus 11R / v13.0
 - OnePlus 12R / v14.0
 - OnePlus 13R / v15.0
 - Pixel 7 / v13.0
 - Galaxy S23 / v13.0

<img width="359" height="773" alt="Image" src="https://github.com/user-attachments/assets/2bd85e95-96d1-475b-9a5c-01b5557e529d" />
<img width="351" height="761" alt="Image" src="https://github.com/user-attachments/assets/3fd4f248-4d4c-4515-9f3a-5ca66ad22695" />


Reproduced on Pixel 6 Emulator with Google Services OS.
Interesting, with open source Android OS it does work !

```
Compat change id reported: 171979766; UID 10146; state: ENABLED
2025-08-21 11:27:21.572  6497-6497  GraphicsEnvironment     network.bisq.mobile.node             V  ANGLE Developer option for 'network.bisq.mobile.node' set to: 'default'
2025-08-21 11:27:21.572  6497-6497  GraphicsEnvironment     network.bisq.mobile.node             V  Neither updatable production driver nor prerelease driver is supported.
2025-08-21 11:27:21.573  6497-6497  NetworkSecurityConfig   network.bisq.mobile.node             D  Using Network Security Config from resource network_security_config debugBuild: false
2025-08-21 11:27:21.574  6497-6497  NetworkSecurityConfig   network.bisq.mobile.node             D  Using Network Security Config from resource network_security_config debugBuild: false
2025-08-21 11:27:21.620  6497-6497  System.out              network.bisq.mobile.node             I  SystemOutFilter: System.out/err completely blocked for release build
2025-08-21 11:27:21.809  6497-6522  libEGL                  network.bisq.mobile.node             D  loaded /vendor/lib64/egl/libEGL_emulation.so
2025-08-21 11:27:21.811  6497-6522  libEGL                  network.bisq.mobile.node             D  loaded /vendor/lib64/egl/libGLESv1_CM_emulation.so
2025-08-21 11:27:21.814  6497-6522  libEGL                  network.bisq.mobile.node             D  loaded /vendor/lib64/egl/libGLESv2_emulation.so
2025-08-21 11:27:21.875  6497-6519  HostConnection          network.bisq.mobile.node             D  createUnique: call
2025-08-21 11:27:21.893  6497-6519  HostConnection          network.bisq.mobile.node             D  HostConnection::get() New Host Connection established 0xb4000077f8908550, tid 6519
2025-08-21 11:27:21.901  6497-6519  HostConnection          network.bisq.mobile.node             D  HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_vulkan_queue_submit_with_commands ANDROID_EMU_sync_buffer_data ANDROID_EMU_vulkan_async_qsri ANDROID_EMU_read_color_buffer_dma GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_0 
2025-08-21 11:27:21.921  6497-6519  OpenGLRenderer          network.bisq.mobile.node             W  Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
2025-08-21 11:27:21.921  6497-6519  OpenGLRenderer          network.bisq.mobile.node             W  Failed to initialize 101010-2 format, error = EGL_SUCCESS
2025-08-21 11:27:21.922  6497-6519  EGL_emulation           network.bisq.mobile.node             D  eglCreateContext: 0xb4000077f8906f90: maj 3 min 0 rcv 3
2025-08-21 11:27:21.925  6497-6519  EGL_emulation           network.bisq.mobile.node             D  eglMakeCurrent: 0xb4000077f8906f90: ver 3 0 (tinfo 0x7a16bfc080) (first time)
2025-08-21 11:27:21.941  6497-6519  Gralloc4                network.bisq.mobile.node             I  mapper 4.x is not supported
2025-08-21 11:27:21.943  6497-6519  HostConnection          network.bisq.mobile.node             D  createUnique: call
2025-08-21 11:27:21.943  6497-6519  HostConnection          network.bisq.mobile.node             D  HostConnection::get() New Host Connection established 0xb4000077f8907290, tid 6519
2025-08-21 11:27:21.944  6497-6519  goldfish-address-space  network.bisq.mobile.node             D  allocate: Ask for block of size 0x100
2025-08-21 11:27:21.944  6497-6519  goldfish-address-space  network.bisq.mobile.node             D  allocate: ioctl allocate returned offset 0x1f3ff8000 size 0x8000
2025-08-21 11:27:21.948  6497-6519  Gralloc4                network.bisq.mobile.node             W  allocator 4.x is not supported
2025-08-21 11:27:21.955  6497-6519  HostConnection          network.bisq.mobile.node             D  HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_vulkan_queue_submit_with_commands ANDROID_EMU_sync_buffer_data ANDROID_EMU_vulkan_async_qsri ANDROID_EMU_read_color_buffer_dma GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_3_0 
2025-08-21 11:27:22.007  6497-6497  isq.mobile.nod          network.bisq.mobile.node             W  Class androidx.compose.runtime.snapshots.SnapshotStateList failed lock verification and will run slower.
                                                                                                    Common causes for lock verification issues are non-optimized dex code
                                                                                                    and incorrect proguard optimizations.
2025-08-21 11:27:22.126  6497-6526  NodeApplic...trapFacade network.bisq.mobile.node             E  Bootstrap: Timeout job cancelled for stage: Starting Bisq
                                                                                                    kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@965447f
2025-08-21 11:27:22.126  6497-6524  NodeApplic...trapFacade network.bisq.mobile.node             E  Bootstrap: Timeout job cancelled for stage: Starting Bisq
                                                                                                    kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@ff79495
2025-08-21 11:27:22.130  6497-6524  isq.mobile.nod          network.bisq.mobile.node             W  Class io.matthewnelson.kmp.tor.runtime.core.EnqueuedJob failed lock verification and will run slower.
2025-08-21 11:27:23.184  6497-6519  EGL_emulation           network.bisq.mobile.node             D  app_time_stats: avg=195.84ms min=2.69ms max=721.13ms count=6
2025-08-21 11:27:23.208  6497-6538  NodeApplic...trapFacade network.bisq.mobile.node             E  Bootstrap: Timeout job cancelled for stage: Launching Tor daemon...
                                                                                                    kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@7f418e2
2025-08-21 11:27:23.229  6497-6561  AndroidApp...ionService network.bisq.mobile.node             E  networkService.initialize() failed (Ask Gemini)
                                                                                                    java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: No virtual method lines()Ljava/util/stream/Stream; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /apex/com.android.art/javalib/core-oj.jar)
                                                                                                    	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:278)
                                                                                                    	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:284)
                                                                                                    	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1629)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at java.lang.Thread.run(Thread.java:920)
                                                                                                    Caused by: java.lang.NoSuchMethodError: No virtual method lines()Ljava/util/stream/Stream; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /apex/com.android.art/javalib/core-oj.jar)
                                                                                                    	at bisq.network.tor.TorService$$ExternalSyntheticApiModelOutline0.m(R8$$SyntheticClass:0)
                                                                                                    	at bisq.network.tor.TorService.readExternalTorConfigMap(TorService.java:328)
                                                                                                    	at bisq.network.tor.TorService.initialize(TorService.java:92)
                                                                                                    	at bisq.network.p2p.node.transport.TorTransportService.initialize(TorTransportService.java:52)
                                                                                                    	at bisq.network.p2p.ServiceNode.getInitializedDefaultNode(ServiceNode.java:270)
                                                                                                    	at bisq.network.p2p.ServiceNodesByTransport.lambda$getInitializedDefaultNodeByTransport$1(ServiceNodesByTransport.java:142)
                                                                                                    	at bisq.network.p2p.ServiceNodesByTransport.$r8$lambda$y4pBO1vy7bTFNomdmryLcUGipuM(ServiceNodesByTransport.java:0)
                                                                                                    	at bisq.network.p2p.ServiceNodesByTransport$$ExternalSyntheticLambda21.get(R8$$SyntheticClass:0)
                                                                                                    	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1627)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
                                                                                                    	at java.lang.Thread.run(Thread.java:920) 
2025-08-21 11:27:23.230  6497-6561  AndroidApp...ionService network.bisq.mobile.node             E  Initializing applicationService failed (Ask Gemini)
                                                                                                    java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: No virtual method lines()Ljava/util/stream/Stream; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /apex/com.android.art/javalib/core-oj.jar)
                                                                                                    	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:278)
                                                                                                    	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:284)
                                                                                                    	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1629)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at java.lang.Thread.run(Thread.java:920)
                                                                                                    Caused by: java.lang.NoSuchMethodError: No virtual method lines()Ljava/util/stream/Stream; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /apex/com.android.art/javalib/core-oj.jar)
                                                                                                    	at bisq.network.tor.TorService$$ExternalSyntheticApiModelOutline0.m(R8$$SyntheticClass:0)
                                                                                                    	at bisq.network.tor.TorService.readExternalTorConfigMap(TorService.java:328)
                                                                                                    	at bisq.network.tor.TorService.initialize(TorService.java:92)
                                                                                                    	at bisq.network.p2p.node.transport.TorTransportService.initialize(TorTransportService.java:52)
                                                                                                    	at bisq.network.p2p.ServiceNode.getInitializedDefaultNode(ServiceNode.java:270)
                                                                                                    	at bisq.network.p2p.ServiceNodesByTransport.lambda$getInitializedDefaultNodeByTransport$1(ServiceNodesByTransport.java:142)
                                                                                                    	at bisq.network.p2p.ServiceNodesByTransport.$r8$lambda$y4pBO1vy7bTFNomdmryLcUGipuM(ServiceNodesByTransport.java:0)
                                                                                                    	at bisq.network.p2p.ServiceNodesByTransport$$ExternalSyntheticLambda21.get(R8$$SyntheticClass:0)
                                                                                                    	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1627)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
                                                                                                    	at java.lang.Thread.run(Thread.java:920) 
2025-08-21 11:27:23.239  6497-6538  NodeApplic...trapFacade network.bisq.mobile.node             E  Bootstrap: Timeout job cancelled for stage: Bootstrap to P2P network
                                                                                                    kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@192dd19
```

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -19,7 +19,8 @@ agp = "8.5.0"
 android-compileSdk = "35"
 android-targetSdk = "35"
 android-minSdk = "24"
-android-node-minSdk = "31"
+# TODO analyse lowering back to API 31 when Bisq2 full J17 support is returned and we can get rid of desugaring
+android-node-minSdk = "33"
 
 # AndroidX Core Libraries
 androidx-core = "1.13.1"

